modules/core/client/components/BoardCredits.js

```diff
@@ -0,0 +1,76 @@
+// import React from 'react';
+import React, { useEffect, useState } from 'react';
+import PropTypes from 'prop-types';
+import { useTranslation } from 'react-i18next';
+import { $on } from '@/modules/core/client/services/angular-compat';
+
+/**
+ * Board Credits component Print out credits for photos used at the page.
+ * Usage: `<BoardsCredits {...props}><WrappedComponent /></Tooltip>`
+ *
+ * @param {object} photos - object containing photo objects accessible by their keys
+ */
+
+/**
+ * @TODO think about transforming photoCredits object structure to JSON
+ */
+
+export default function BoardCredits({ photoCredits: initialPhotoCredits }) {
+  const { t } = useTranslation('messages');
+
+  const [photoCredits, setPhotoCredits] = useState(initialPhotoCredits); // TODO: get initial value from app.photoCredits
+
+  useEffect(() => {
+    return () => {
+      $on('photoCreditsUpdated', (scope, photo) => {
+        setPhotoCredits({ ...photoCredits, ...photo });
+      });
+
+      $on('photoCreditsRemoved', (scope, photo) => {
+        setPhotoCredits({ ...photoCredits, ...photo });
```

Comment:
This needs to _remove_ the photo. Similar to the existing handler for the core part https://github.com/Trustroots/trustroots/blob/6bdf0f9ca4243e6a4c134782aba53177e091c489/modules/core/client/controllers/app.client.controller.js#L311-L316

Maybe a nice modern approach (using https://lodash.com/docs/4.17.15#omit) would be:

```js
// at top...
import omit from 'lodash/omit';

// in $on callback handler
setPhotoCredits(omit(photoCredits, Object.keys(photo));
```

I didn't try it, but I think that should work :)